app/charts/shared/chart-helpers.tsx

```diff
@@ -87,48 +90,31 @@ export const usePreparedData = ({
   const { categories } = interactiveFilters;
   const activeInteractiveFilters = Object.keys(categories);
 
-  const preparedData = useMemo(() => {
-    if (!timeFilterActive && !legendFilterActive) {
-      return sortedData;
-    } else if (timeFilterActive && !legendFilterActive && getX) {
-      return from && to
-        ? sortedData.filter(
-            (d) =>
-              from &&
-              to &&
-              getX(d).getTime() >= from.getTime() &&
-              getX(d).getTime() <= to.getTime()
-          )
-        : sortedData;
-    } else if (!timeFilterActive && legendFilterActive && getSegment) {
-      return sortedData.filter(
-        (d) => !activeInteractiveFilters.includes(getSegment(d))
-      );
-    } else if (timeFilterActive && legendFilterActive && getX && getSegment) {
-      return from && to && activeInteractiveFilters
-        ? sortedData
-            .filter(
-              (d) =>
-                from &&
-                to &&
-                getX(d).getTime() >= from.getTime() &&
-                getX(d).getTime() <= to.getTime()
-            )
-            .filter((d) => !activeInteractiveFilters.includes(getSegment(d)))
-        : sortedData;
-    } else {
-      return sortedData;
-    }
+  const allFilters = useMemo(() => {
+    const timeFilter: ValuePredicate | null =
+      getX && from && to && timeFilterActive
+        ? (d: any) =>
+            getX(d).getTime() >= from.getTime() &&
+            getX(d).getTime() <= to.getTime()
+        : null;
+    const legendFilter: ValuePredicate | null =
+      legendFilterActive && getSegment
+        ? (d: any) => !activeInteractiveFilters.includes(getSegment(d))
+        : null;
+    return flowRight.apply(null, [legendFilter, timeFilter].filter(truthy));
```

Comment:
This doesn't really matter, but I think you could do `(d: Observation)` here?

Reply:
Yes, done, thanks !